src/routes/ParentPaymentSheet.res

```diff
@@ -21,22 +21,29 @@ let make = () => {
   let setConfirmButtonDataRef = React.useCallback1(confirmButtonDataRef => {
     setConfirmButtonDataRef(_ => confirmButtonDataRef)
   }, [setConfirmButtonDataRef])
+  let samsungPayValidity = SamsungPay.useSamsungPayValidityHook()
 
   <FullScreenSheetWrapper>
-    {switch (allApiData.savedPaymentMethods, allApiData.additionalPMLData.paymentType) {
-    | (_, None)
-    | (Loading, _) =>
-      nativeProp.hyperParams.defaultView
+    {switch (
+      allApiData.savedPaymentMethods,
+      allApiData.additionalPMLData.paymentType,
+      samsungPayValidity,
+    ) {
+    | (_, _, SamsungPay.Checking) => <SdkLoadingScreen />
+    | (_, _, SamsungPay.Not_Started) => <SdkLoadingScreen />
+    | (_, None, _)
+    | (Loading, _, _) =>
+      nativeProp.hyperParams.defaultView && samsungPayValidity->SamsungPay.isSamsungPayValid
         ? <PaymentSheet setConfirmButtonDataRef />
```

Comment:
Should we move this to PMListModifier.res as checks for other wallets are also handled there